### PR TITLE
AUT-4907 Helm chart for Apigee authorizer

### PR DESCRIPTION
**Pull request by djoo-ce**
## Description
Helm chart for ApigeeX and Apigee Edge authorizer

## Type of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Tests (extending the test suite)
- [ ] Refactor (internal improvement that doesn't change product functionality)
- [ ] Other (if none of the other choices apply)

### CODE REVIEW

**Review by sburke-ce — APPROVED** [review]

File: charts/apigee-authorizer/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                secretKeyRef:
+                  name: {{ include "apigee-authorizer.clientCredentialsName" . }}
+                  key: clientSecret
+            - name: ACP_ISSUER_URL
+              value: "{{ .Values.issuerURL }}"
+            - name: ACP_TENANT_ID
+              value: "{{ .Values.tenantID }}"
+            - name: ACP_SERVER_ID
+              value: "{{ .Values.serverID }}"
+          {{- range $key, $value := .Values.extraEnv }}
```
Comment (ppodbielski-cloudentity) [inline charts/apigee-authorizer/templates/deployment.yaml:53]: @holowinski do we want to pass args here?

File: charts/apigee-authorizer/values.yaml [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  #    hosts:
+  #      - apigee.domain.com
+
+
+## Additional apigee-authorizer ConfigMap entries
+##
+data:
+  ## Example self-signed cert used to serve HTTPS.
+  ##
+  cert.pem: |
```
Comment (ppodbielski-cloudentity) [inline charts/apigee-authorizer/values.yaml:126]: @holowinski do you want to move this to /file like in ACP?

File: charts/apigee-authorizer/values.yaml [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    c+F8iTx0R9P2VZ2P1PlK5UovNDO8SXKhim7cZljM7IL1xnndlIQTua/sY8pTD4SZ
+    Hg+EgtbwvUGoinVsAXiBQr8L1+4k+G7FQ7C02n5nYi/nICjrtjm+mjkHTqwquz50
+    bp+CU+6TEJMRSEsrIk4YzQKBgQChzHB/3hO8+96oNCScb70uP5SQury1oYMqWJge
+    ENYDqZ8t7ZUY7ljJyZBuitkgNLK0p4m6keKLcdYtFd9fbQkIxv3SoAzQqmKhkS1n
+    0YtZUGXyOMWzk98h5rrajCiGF+FVUCm2OriGNNPJZPtiljW3Py4W7sVnDiU/ikCq
+    /7CNqQKBgQCL1lk7yQ572D8Sw+Z+sOc4Tk5aO9i394RE2DX3X810ZoAN4Q/1X61J
+    64/OUsD3B2FF+WhkjsJAlLBYkVbJUfjY4OZywpnScLjJYTysYlrAArz+5RYEwYUi
+    ulKp5/FbjejQNCJr6FD7xouVXUa1yNP8DMNEnhYI7rDGLqIMwEaVRw==
+    -----END RSA PRIVATE KEY-----
+  ## Example Google Service account access key. Needed only for ApigeeX.
```
Comment (ppodbielski-cloudentity) [inline charts/apigee-authorizer/values.yaml:182]: I think we dont need here example configuration. Refer to reference.yaml and instruction

**Review by ppodbielski-cloudentity — CHANGES_REQUESTED** [review]

File: charts/apigee-authorizer/templates/deployment.yaml (hunk 1 above)
Comment (djoo-ce) [reply charts/apigee-authorizer/templates/deployment.yaml:53]: @holowinski @ppodbielski-cloudentity ApigeeX requires an environment variable `GOOGLE_APPLICATION_CREDENTIALS` which does not have a corresponding field in the `reference.yaml`. That's the reason for "extraEnv" section in this chart.

File: charts/apigee-authorizer/values.yaml (hunk 3 above)
Comment (djoo-ce) [reply charts/apigee-authorizer/values.yaml:182]: There is no information about the Google service account key in reference.yaml right now.
Information about it is only in Authz docs (https://docs.authorization.cloudentity.com/guides/developer/protect/apigee/apigeex/?q=apigee). Will the info in the docs be enough for users, and we can go ahead remove this example?

**Review by mbilski — CHANGES_REQUESTED** [review]
Please update this helm chart to match istio-authorizer helm chart.
There were lots of changes recently.

**Pull request closed without merging**